app/gui/src/project-view/components/MarkdownEditor/markdown/inlineFormatting/markdownDocument.ts

```diff
@@ -0,0 +1,299 @@
+import { TextDocument } from '@/components/MarkdownEditor/markdown/inlineFormatting/textDocument'
+import {
+  getUnformattableAncestor,
+  isDelimiter,
+  pointFormatAncestorInfo,
+  splitNodesAt,
+  splitRange,
+  trimRangeDelimiters,
+} from '@/components/MarkdownEditor/markdown/inlineFormatting/trees'
+import {
+  andFormatting,
+  depthsToStates,
+  type FormatDepths,
+  type FormatNode,
+  type FormatStates,
+  nodeMarkToken,
+  type NormalizedRange,
+  type Range,
+} from '@/components/MarkdownEditor/markdown/inlineFormatting/types'
+import { type Text } from '@codemirror/state'
+import { type Tree } from '@lezer/common'
+import { type DeepReadonly } from 'vue'
+
+interface Side {
+  fromOrTo: 'from' | 'to'
+}
```

Comment:
Any particular reason, why is it interface instead of just `type Side = 'from' | 'to'`?